en/services/ftp.md

```diff
@@ -209,46 +209,49 @@ On completion of the burst (or the file), if there are any missing parts of the
-    Note right of Client: Read any missing chunks using ReadFile at offset
-    Client- >>Server:  ReadFile(session, size, offset)
-    Server-- >>Client: ACK(session, size=len(buffer), data[0]=buffer)
-    Note right of Client: Close session when whole data file collected
-    Client- >>Server:  TerminateSession(session)
-    Server-- >>Client: ACK()
+    Note over Client,Server: Open file (for burst reading)
+    Note right of Client: Request open file
+    Client->>Server:  OpenFileRO( data[0]=path, size=len(path) )
+    Server->>Client: ACK( session, size=4, data=len(file) )
+    Note over Client,Server: Read the file in bursts (repeat sequence below at new offsets)
+    Note right of Client: Request burst read part of the file (in this case at offset=0)
+    Client->>Server:  BurstReadFile( session, offset=0, size=data_per_burst )
+    Note left of Server: Server streams data in chunks at increasing offsets with burst_complete=0
+    Server->>Client: ACK(session, burst_complete=0, offset=0, size=len(buffer), data[0]=buffer)
+    Server->>Client: ACK(session, burst_complete=0, offset=<current_offet>, size=len(buffer), data[0]=buffer)
+    Server->>Client: ...
+    Note left of Server: For last message in burst set burst_complete=1
+    Server->>Client: ACK(session, burst_complete=1, offset=<offset_of_last_burst_chunk>, size=len(buffer), data[0]=buffer)
+    Note over Client,Server: Read any missing parts using BurstReadFile (sequence above) or ReadFile
+    Note over Client,Server: Close session when whole file collected
+    Client->>Server:  TerminateSession(session)
+    Server->>Client: ACK()
 -->
 
 The sequence of operations is:
 1. GCS (client) sends [OpenFileRO](#OpenFileRO) command specifying the file path to open.
    - The payload must specify: `data[0]`= file path string, `size`=length of file path string.
-1. Drone (server) responds with either 
+1. Drone (server) responds with either
    - ACK on success. The [payload](#payload) must specify fields: `session` = file session id, `size` = 4, `data` = length of file that has been opened.
-   - NAK with [error information](#error_codes), e.g. `NoSessionsAvailable`, `FileExists`. 
+   - NAK with [error information](#error_codes), e.g. `NoSessionsAvailable`, `FileExists`.
      The GCS may cancel the operation, depending on the error.
 1. Client (i.e. GCS) sends [BurstReadFile](#BurstReadFile) command specifying the part of the file that it wants to get.
    - The payload must specify: `session`: the current session id, `offset` = offset in file of start of burst, `data`= length of burst, `size`=4.
-1. Server (drone) responds with either 
-   - ACK on success. The [payload](#payload) must specify fields: `session` = file session id, `size` = 4, `data` = lenght of file in burst. 
+1. Server (drone) responds with either
+   - ACK on success. The [payload](#payload) must specify fields: `session` = file session id, `size` = 4, `data` = lenght of file in burst.
```

Comment:
So to be clear `data` is from offset to end of file?

```suggestion
   - ACK on success. The [payload](#payload) must specify fields: `session` = file session id, `size` = 4, `data` = length of file in burst.
```

Reply:
No, data is from offset to end of file, or end of payload buffer.